### DISCUSSION TITLE
Can the EMSModelServerModule configuration support multiple semantic file extensions?

### DISCUSSION THREAD

**Discussion by RobMoloney** [eclipse-emfcloud/emfcloud#232] [ModelServer]
We are developing an application that uses multiple file extensions. Can I register one instance of the EMSModelServerModule and handle multiple semantic file extensions?

**Comment by ndoschek** [comment]
Hi @RobMoloney,

Thank you for interest in EMF cloud! :tada: 

Regarding your question about handling multiple file extensions with one instance of the `EMSModelServerModule`, it depends on a few factors. If your diagram language uses multiple file extensions, GLSP could handle that scenario and you could customize the Model Server to also accept multiple extensions.
However, if the diagram language and the semantic model are different, I would suggest using modules per language/file extension.

Additionally, please note that the model server projects are currently in maintenance mode and not under active development.

Depending on your plans, the project template of GLSP that works with an EMF-based source model might also be interesting for you: [GLSP EMF-based project template](https://github.com/eclipse-glsp/glsp-examples/tree/master/project-templates/java-emf-theia).

Please let me know if you have any additional questions.

All the best,
Nina
